pkg/controller/endpointslicemirroring/endpointslicemirroring_controller.go

```diff
@@ -273,6 +278,36 @@ func (c *Controller) syncEndpoints(key string) error {
 		return err
 	}
 
+	endpoints, err := c.endpointsLister.Endpoints(namespace).Get(name)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return c.deleteMirroredSlices(namespace, name)
+		}
+		return err
+	}
+
+	if !c.shouldMirror(endpoints) {
+		return c.deleteMirroredSlices(namespace, name)
+	}
+
+	svc, err := c.serviceLister.Services(namespace).Get(name)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.V(4).Infof("Skipping mirroring for %s/%s due to missing Service", namespace, name)
+			return c.deleteMirroredSlices(namespace, name)
+		}
+		klog.Errorf("Error fetching %s/%s Service: %v", namespace, name, err)
+		return err
+	}
+
+	// This means that if a Service transitions away from a nil selector, any
+	// mirrored EndpointSlices will not be cleaned up. This matches the behavior
+	// of the Endpoints and EndpointSlice controllers.
+	if svc.Spec.Selector != nil {
+		klog.V(4).Infof("Not mirroring EndpointSlice for %s/%s because selector is not nil", namespace, name)
+		return nil
```

Comment:
> This matches the behavior of the Endpoints and EndpointSlice controllers.

do you mean it is the inverse of the behavior of those controllers? (e.g. those controllers wouldn't clean up an endpoints/endpointsslice object if a service switched from a selector to a nil selector)?

Reply:
this also seems like a comment that should have an associated issue link to clean up this behavior for all three controllers

Reply:
Yeah that was very poorly worded. I meant to say that they all do not clean up resources during a transition to/from a nil selector. Instead I've added a link to https://github.com/kubernetes/kubernetes/issues/91072.